include/app_memory/app_memdomain.h

```diff
@@ -82,36 +82,52 @@ struct app_region {
 #endif	/* CONFIG_MPU_REQUIRES_POWER_OF_TWO_ALIGNMENT */
 
 #define appmem_partition(name) \
-	extern char *data_smem_##name; \
-	extern char *data_smem_##name##b; \
-	smem_size_declare(name);		  \
 	_app_dmem_pad(name) char name##_dmem_pad; \
 	_app_bmem_pad(name) char name##_bmem_pad; \
-	__kernel struct k_mem_partition mem_domain_##name; \
-	__kernel struct app_region name; \
+	__kernel struct k_mem_partition mem_partition_##name; \
+	__kernel struct app_region name;
+
+#define appmem_partition_header(name) \
+	extern char *data_smem_##name;	  \
+	extern char *data_smem_##name##b; \
+	smem_size_declare(name); \
+	extern struct k_mem_partition mem_partition_##name; \
+	extern struct app_region name; \
 	static inline void appmem_init_part_##name(void) \
 	{ \
 		name.dmem_start = (char *)&data_smem_##name; \
 		name.bmem_start = (char *)&data_smem_##name##b; \
 		smem_size_assign(name);				\
 		sys_dlist_append(&app_mem_list, &name.lnode); \
-		mem_domain_##name.start = (u32_t) name.dmem_start; \
-		mem_domain_##name.attr = K_MEM_PARTITION_P_RW_U_RW; \
-		name.partition = &mem_domain_##name; \
+		mem_partition_##name.start = (u32_t) name.dmem_start; \
+		mem_partition_##name.attr = K_MEM_PARTITION_P_RW_U_RW; \
+		name.partition = &mem_partition_##name; \
 	}
 
 /* A helper function that should be called to define multiple partitions*/
-#define APPMEM_PARTITION_DEFINE(...)\
+#define APPMEM_PARTITION_OBJECT_DEFINE(...)\
```

Comment:
Please fix your git history.
You defined this to one thing in the previous patch, now you are changing it